GoldRush.js

```diff
@@ -0,0 +1,121 @@
+class GoldRush extends Matrix{
+    constructor(numRows, numColumns ){
+        super(numRows, numColumns)
+        this.player1 = {
+            mark:1,
+            score:0,
+            x:0,
+            y:0
+        }
+        this.player2 = {
+            mark:2,
+            score:0,
+            x: numRows-1,
+            y: numColumns-1
+        }
+        this.freeCells  = (this.numColumns * this.numRows) -2
+        this.numCoins = 10
+        this.numWalls = 6
+        this.coins = []
```

Comment:
I think there is a lot of duplication of data here. Where you have the coins, and then only later you place them. I'm thinking you should generate the locations for the coins and place them at the exact same time. Then you don't have to store them in multiple places.